cmu-db/noisepage
# Support server cmd

Summary of Changes

Jenkinsfile
- Added new tests for WAL disabled and parallel execution disabled in End-to-End Debug
- Added new tests for WAL disabled and WAL write to RAM disk to End-to-End Performance
- Added new TPCC tests to TPCC Performance
- Disabled Microbenchmark tests

Jenkinsfile-nightly
- Added tests with WAL is written to RAM disk
- Added tests that mimic the tests run in the normal build pipeline End-to-End Performance stage
- Disable Microbenchmark tests

script/testing/oltpbench/configs/README.md
- Update instructions to include server_args

script/testing/oltpbench/configs/end_to_end_debug/…
- Switch all End-tp-End Debug tests to use 2 terminals (except TPCC tests)
- Add WAL disabled test
- Add a metrics_pipeline and use_query_cache test
- Add a tests with parallel execution off

script/testing/oltpbench/configs/end_to_end_performance/…
- Add connection thread count = 32 to all tests
- Add TPCC tests
- Change all TATP tests to 16 terminals and TPCC tests to 32 terminals
- Add WAL disabled test
- Add WAL to RAM disk test

script/testing/oltpbench/configs/nightly/nightly.json
- Add connection thread count = 32
- Add continue on error flag
- Add environment details about where the WAL is written
- Add 32 terminals option to all benchmark types
- Change the 600 second test. To 32 terminals

script/testing/oltpbench/configs/nightly/nightly_ramdisk.json
- Add nightly tests that measure performance when WAL is written to RAM disk
- These tests mimic script/testing/oltpbench/configs/nightly/nightly.json 

script/testing/oltpbench/configs/nightly/nightly_wal_disabled.json
- Added tests that mimic the WAL disabled tests from the normal build pipeline for comparison

script/testing/oltpbench/run_oltpbench.py
- General refactoring for readability + modularity
- Add logging.shutdown() to make sure stdout is flushed before exiting

script/testing/util/test_server.py
- General refactoring for readability + modularity

src/include/settings/settings_defs.h
- Increase max buffer segment pool size to 100000000

src/traffic_cop/traffic_cop.cpp
- Change Query failed error to ERRCODE_T_R_SERIALIZATION_FAILURE

Terrier Build: http://jenkins.db.cs.cmu.edu:8080/blue/organizations/jenkins/testing-team%2Fterrier/detail/support_server_cmd_dan/39/pipeline

Nightly Build: http://jenkins.db.cs.cmu.edu:8080/blue/organizations/jenkins/testing-team%2Fterrier-nightly/detail/support_server_cmd_dan/22/pipeline
